hyperspy/events.py

```diff
@@ -90,22 +187,28 @@ def connect(self, function, nargs='all'):
         self._connected[nargs].add(function)
 
     def disconnect(self, function):
+        """
+        Disconnects a function from the event. The passed function will be
+        disconnected irregardless of which 'nargs' argument was passed to
+        connect().
+        """
         for c in self._connected.itervalues():
             if function in c:
                 c.remove(function)
 
-    def trigger(self, *args, **kwargs):
-        if not self.suppress:
-            for nargs, c in self._connected.iteritems():
+    def trigger(self, *args):
+        if not self._suppress:
+            # Loop on copy to deal with callbacks which change connections
+            for nargs, c in self._connected.copy().iteritems():
                 if nargs is 'all':
                     for f in c:
-                        f(*args, **kwargs)
```

Comment:
From the commit message: "Remove for now, as it makes bahvior inconsistent. Can readd later if we come across a case where we need it."

I basically feel that it mixes badly with the nargs portion, but am not 100% myself. Do you think I should put it back in?